projects/plugins/jetpack/modules/widget-visibility/editor/index.jsx

```diff
@@ -1,9 +1,394 @@
+					...rules.slice( i + 1 ),
+				];
+				setAttributes( {
+					conditions: {
+						...maybeAddDefaultConditions( conditions ),
+						rules: newRules,
+					},
+				} );
+			},
+			[ setAttributes, conditions, rules ]
+		);
+
+		const setMinor = useCallback(
+			( i, value ) => {
+				// Don't allow section headings to be set
+				if ( value && value.includes( '__HEADER__' ) ) {
+					return;
+				}
+				const newRules = [
+					...rules.slice( 0, i ),
+					{ ...rules[ i ], minor: value },
+					...rules.slice( i + 1 ),
+				];
+				setAttributes( {
+					conditions: {
+						...maybeAddDefaultConditions( conditions ),
+						rules: newRules,
+					},
+				} );
+			},
+			[ setAttributes, conditions, rules ]
+		);
+
+		let mainRender = null;
+		if ( rules.length === 0 ) {
+			mainRender = (
+				<BaseControl
+					id="widget-vis__wrapper"
+					className="widget-vis__wrapper"
+					label={ __( 'Visibility', 'jetpack' ) }
+					help={ __(
+						'No visibility rules yet. Add at least one rule to use this feature.',
+						'jetpack'
+					) }
+				>
+					<Button isSecondary onClick={ addNewRule } className="widget-vis__add-new-rule">
+						{ __( 'Add new rule', 'jetpack' ) }
+					</Button>
+				</BaseControl>
+			);
+		} else {
+			mainRender = (
+				<BaseControl
+					className="widget-vis__wrapper"
+					id="widget-vis__wrapper"
+					label={ __( 'Visibility', 'jetpack' ) }
+				>
+					<SelectControl
+						className="widget-vis__show-hide"
+						label={ __( 'Action', 'jetpack' ) }
+						hideLabelFromVision
+						value={ attributes.action }
+						options={ [
+							{ label: __( 'Show this block', 'jetpack' ), value: 'show' },
+							{ label: __( 'Hide this block', 'jetpack' ), value: 'hide' },
+						] }
+						onChange={ setAction }
+					/>
+					{ rules.map( ( rule, i ) => (
+						<VisibilityRule
+							key={ i }
+							rule={ rule }
+							i={ i }
+							onDelete={ () => deleteRule( i ) } // eslint-disable-line react/jsx-no-bind
+							setMajor={ value => setMajor( i, value ) } // eslint-disable-line react/jsx-no-bind
+							setMinor={ value => setMinor( i, value ) } // eslint-disable-line react/jsx-no-bind
+						/>
+					) ) }
+					{ rules.length > 1 && (
+						<ToggleControl
+							className="widget-vis__match-all"
+							label={ __( 'Match all rules', 'jetpack' ) }
+							checked={ conditions.match_all === '1' }
+							onChange={ toggleMatchAll }
+						/>
+					) }
+					<Button isSecondary onClick={ addNewRule }>
+						{ __( 'Add new rule', 'jetpack' ) }
+					</Button>
+				</BaseControl>
+			);
+		}
+
+		return (
+			<Fragment>
+				<BlockEdit { ...props } />
+				{ isSelected && blockHasVisibilitySettings( props.name ) && (
+					<InspectorAdvancedControls>{ mainRender }</InspectorAdvancedControls>
+				) }
+			</Fragment>
+		);
+	},
+	'visibilityAdvancedControls'
+);
+
+wp.hooks.addFilter( 'editor.BlockEdit', 'widget/visibility', visibilityAdvancedControls );
```

Comment:
Here and above, you could have added an import for `addFilter` at the top of the file, it may be easier to quickly go through?

That's not a blocker though.